src/lib.rs

```diff
@@ -116,6 +116,16 @@ mod debug;
 mod serde;
 pub(crate) mod utils;
 
+// Make formattable by rustfmt.
```

Comment:
So just to confirm: this `cfg` is always evaluated to `false` but at least it allows `rustfmt` to see the file. That's a super clever trick. Thanks for that!

Reply:
That's right.

In the longer term, hopefully rustfmt will get a built-in understanding of the syntax of `cfg_if!` (https://github.com/rust-lang/rustfmt/issues/3253) or `cfg_match!` (https://github.com/rust-lang/rust/issues/115585) and the workaround won't be needed.

Or alternatively, someone might make an alternative macro that doesn't rely on dedicated rustfmt support for its syntax, something like:

```rust
#[apply_cfg]
enum __ {
    r#if = #[cfg(feature = "unknown-ci")]
    {
        // This is used in CI to check that the build for unknown targets is compiling fine.
        mod unknown;
        use crate::unknown as sys;

        #[cfg(test)]
        pub(crate) const MIN_USERS: usize = 0;
    },

    r#else = #[cfg(any(
        target_os = "macos",
        target_os = "ios",
        target_os = "linux",
        target_os = "android",
        target_os = "freebsd"
    ))]
    {
        mod unix;
        use crate::unix::sys;

        #[cfg(feature = "network")]
        mod network;
        #[cfg(feature = "network")]
        use crate::unix::network_helper;

        #[cfg(test)]
        pub(crate) const MIN_USERS: usize = 1;
    },

    ...
}
```

Reply:
I prefer the current solution. It duplicates code but at least it's "easy" to see what's going on. Gonna check with rustfmt team if they'd be open to add some special cases for macros like `cfg_if!`.